test/CliMenuTest.php

```diff
@@ -480,6 +480,66 @@ public function testAddCustomControlMapping() : void
         static::assertStringEqualsFile($this->getTestFile(), $this->output->fetch());
     }
 
+    public function testAddCustomControlMappingsThrowsExceptionWhenOverwritingExistingDefaultControls() : void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Cannot rebind this input');
+
+        $menu = new CliMenu('PHP School FTW', []);
+        $menu->addCustomControlMappings([
+            ' ' => function () {
+            }
+        ]);
+    }
+
+    public function testAddCustomControlMappingsThrowsExceptionWhenAttemptingToOverwriteAddedCustomControlMap() : void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Cannot rebind this input');
+
+        $menu = new CliMenu('PHP School FTW', []);
+        $menu->addCustomControlMappings([
+            'c' => function () {
+            },
+            'c' => function () {
+            }
+        ]);
+    }
+
+    public function testAddCustomControlMappings() : void
+    {
+        $first = true;
+        $this->terminal->expects($this->any())
+            ->method('read')
+            ->will(
```

Comment:
Here you can just use `->willReturn('c', 'x')`. `c` will be returned on the first call to `read` and `x` on the next :)

Reply:
No ! You've got to be kinding me. What will it return on subsequent calls ? 'x' ?

Reply:
I think it just returns null after anything you don't specify.